src/ErrorSparseTask.cpp

```diff
@@ -107,7 +111,8 @@ void ErrorSparseTask::error_response() {
   }
 }
 
-void ErrorSparseTask::run(const Configuration& config) {
+void ErrorSparseTask::run(const Configuration& config, bool testing) {
+  s3_initialize_aws();
```

Comment:
I suggest we put an s3_initialize_aws() before run_tasks in parameter_server.cpp and remove this.